### DISCUSSION TITLE
Sending a bulk email using phpmailer

### DISCUSSION THREAD

**Discussion by dhen22** [PHPMailer/PHPMailer#2816] [Q&A] [answered]
hi, i really need help i used phpmailer and it's actually working fine it send to a multiple recipients but i have this kind a error : Fatal error: Uncaught PHPMailer\PHPMailer\Exception: You must provide at least one recipient email address. in C:\xampp\htdocs\Email_php\vendor\phpmailer\phpmailer\src\PHPMailer.php:1553 Stack trace: #0 C:\xampp\htdocs\Email_php\vendor\phpmailer\phpmailer\src\PHPMailer.php(1496): PHPMailer\PHPMailer\PHPMailer->preSend() #1 C:\xampp\htdocs\Email_php\email.php(49): PHPMailer\PHPMailer\PHPMailer->send() #2 {main} thrown in C:\xampp\htdocs\Email_php\vendor\phpmailer\phpmailer\src\PHPMailer.php on line 1553

**Comment by Synchro** [comment, marked as answer]
The "You must provide at least one recipient email address." error means that at the point when you ask it to send, it's not configured to send to anyone. We can't guess what your code is doing without seeing it, and that's where the error will be.

**Reply by dhen22** [reply]
`````````````````````````````````````````````````````````````````````````````
for($i=0; $i<$totalEmails; $i++){

	$mail->From = "";
	$from = "";
	$mail->FromName = $from;

	$Email = $emailArray[$i];
	$mail->AddReplyTo($Email, $from);

	$to = $Email;
	$mail->AddAddress($to, "");

	$mail->IsHTML(true);
	$mail->Subject = '!';

	$mail->Body = $bodyArray[$i];
	$mail->Send();

	$mail->ClearAddresses();
}

// it never meet the IF STATEMENT i think because of my ClearAddresses();

if(!$mail->send()){

    echo "Mailer Error:". $mail->ErrorInfo;
}else{

    echo "Message has been send";
}
?>
````````````````````````````````````````````````````````````````````````````

do you have any idea whats wrong on my code ?

**Reply by Synchro** [reply]
You're calling `send` twice, but the second time it's after you have called `clearAddresses`, so there will be no addresses to send it to. I suggest you base your code on [the mailing list example provided with PHPMailer](https://github.com/PHPMailer/PHPMailer/blob/master/examples/mailing_list.phps). You only need to call `send` *once* for each message you want to send.

**Reply by dhen22** [reply]
I get it now sir ! thanks a lot ! you saved me, one more question can i use a text file that have the emails recipients and turn it into arrays then implement it to my code ? give me ideas sir, thank you i don't have any idea in php.

**Reply by Synchro** [reply]
Sounds like you had a perfectly reasonable idea there! PHPMailer doesn't care where your addresses come from – it's most common to use a database or data from a form submission, but a text file will work fine too.
